interfaces.go

```diff
@@ -5,14 +5,31 @@
 package optimize
 
 // A Method can optimize an objective function.
+//
+// It uses a reverse-communication interface design represented by the Init()
+// and Iterate() methods to pass and retrieve information from the client. Its
+// aim is to isolate the method from user-supplied information and data and
+// their representation, and from common operations like checking for (various
+// types of) convergence and maintaining statistics.
+//
+// When Method needs the client to perform an operation like function
+// evaluation or check for convergence, it fills the Location with data and
+// returns a corresponding Operation. The result then can be retrieved from
+// Location upon the next call to Iterate().
+//
+// Methods must not return InitIteration and PostIteration operations. These
+// are reserved for the clients to be passed to Recorders.
 type Method interface {
-	// Init initializes the method and stores the first location to evaluate
-	// in xNext.
-	Init(loc *Location, xNext []float64) (EvaluationType, IterationType, error)
+	// Init initializes the method based on the initial data in loc, updates it
```

Comment:
Like I wrote above, Method can set also other fields of Loc when it is asking for MajorIteration. That's why I didn't write just X.